crates/scroll/node/src/node.rs

```diff
@@ -0,0 +1,271 @@
+use reth_ethereum_engine_primitives::EthEngineTypes;
+use reth_ethereum_forks::EthereumHardforks;
+use reth_evm::execute::BasicBlockExecutorProvider;
+use reth_network::PeersInfo;
+use reth_node_builder::{
+    components::{
+        Components, ComponentsBuilder, ConsensusBuilder, ExecutorBuilder, NetworkBuilder,
+        PayloadServiceBuilder, PoolBuilder,
+    },
+    BuilderContext, FullNodeTypes, Node,
+};
+use reth_node_types::{NodePrimitives, NodeTypes, NodeTypesWithEngine};
+use reth_payload_builder::PayloadBuilderHandle;
+use reth_primitives::BlockBody;
+use reth_provider::{
+    providers::ChainStorage, BlockBodyReader, BlockBodyWriter, ChainSpecProvider, DBProvider,
+    EthStorage, ProviderResult, ReadBodyInput,
+};
+use reth_scroll_evm::{ScrollEvmConfig, ScrollExecutionStrategyFactory};
+use reth_tracing::tracing::info;
+use reth_transaction_pool::{noop::NoopTransactionPool, TransactionPool};
+use reth_trie_db::MerklePatriciaTrie;
+
+// #################### NODE ####################
+
+/// The Scroll node implementation.
+#[derive(Clone, Debug)]
+pub struct ScrollNode;
+
+impl<N> Node<N> for ScrollNode
+where
+    N: FullNodeTypes,
+    // TODO (scroll): replace with `ScrollChainSpec`.
+    N::Types: NodeTypes<ChainSpec = ChainSpec, Primitives = ScrollPrimitives>,
+{
+    type ComponentsBuilder = ComponentsBuilder<
+        N,
+        ScrollPoolBuilder,
+        ScrollPayloadBuilder,
+        ScrollNetworkBuilder,
+        ScrollExecutorBuilder,
+        ScrollConsensusBuilder,
+    >;
+    type AddOns = ();
+
+    fn components_builder(&self) -> Self::ComponentsBuilder {
+        ComponentsBuilder::default()
+            .node_types::<N>()
+            .pool(ScrollPoolBuilder)
+            .payload(ScrollPayloadBuilder)
+            .network(ScrollNetworkBuilder)
+            .executor(ScrollExecutorBuilder)
+            .consensus(ScrollConsensusBuilder)
+    }
+
+    fn add_ons(&self) -> Self::AddOns {}
+}
+
+// #################### NODE POOL ####################
+
+/// Pool builder for Scroll.
+#[derive(Debug)]
+pub struct ScrollPoolBuilder;
+
+impl<Node> PoolBuilder<Node> for ScrollPoolBuilder
+where
+    Node: FullNodeTypes,
+{
+    type Pool = NoopTransactionPool;
+
+    async fn build_pool(self, _ctx: &BuilderContext<Node>) -> eyre::Result<Self::Pool> {
+        Ok(NoopTransactionPool::default())
+    }
+}
+
+// #################### NODE PAYLOAD ####################
+
+/// Payload builder for Scroll.
+#[derive(Debug)]
+pub struct ScrollPayloadBuilder;
+
+impl<Node, Pool> PayloadServiceBuilder<Node, Pool> for ScrollPayloadBuilder
+where
+    Node: FullNodeTypes,
+    Pool: TransactionPool,
+{
+    async fn spawn_payload_service(
+        self,
+        _ctx: &BuilderContext<Node>,
+        _pool: Pool,
+    ) -> eyre::Result<
+        PayloadBuilderHandle<<<Node as FullNodeTypes>::Types as NodeTypesWithEngine>::Engine>,
+    > {
+        // TODO (scroll): trying to send on the channel will return an error.
```

Comment:
question: Should we create a simple service that receives on the channel and logs that the payload builder service is disabled?

Reply:
I was thinking about that too. will add it!